src/silo/test/pango_lineage_filter.test.cpp

```diff
@@ -0,0 +1,116 @@
+#include <nlohmann/json.hpp>
+
+#include <optional>
+
+#include "silo/test/query_fixture.test.h"
+
+using silo::PangoLineageAliasLookup;
+using silo::ReferenceGenomes;
+using silo::config::DatabaseConfig;
+using silo::config::ValueType;
+using silo::test::QueryTestData;
+using silo::test::QueryTestScenario;
+
+static const std::string SOME_BASE_PANGO_LINEAGE = "P.1";
+static const std::string SOME_SUBLINEAGE = "A.1";
+
+nlohmann::json createDataWithPangoLineageValue(const std::string& primaryKey, std::string value) {
+   return {
+      {"metadata", {{"primaryKey", primaryKey}, {"pango_lineage", value}}},
+      {"alignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"unalignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"alignedAminoAcidSequences", {{"gene1", nullptr}}}
+   };
+}
+
+nlohmann::json createDataWithPangoLineageNullValue(const std::string& primaryKey) {
+   return {
+      {"metadata", {{"primaryKey", primaryKey}, {"pango_lineage", nullptr}}},
+      {"alignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"unalignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"alignedAminoAcidSequences", {{"gene1", nullptr}}}
+   };
+}
+const std::vector<nlohmann::json> DATA = {
+   createDataWithPangoLineageValue("id_0", SOME_BASE_PANGO_LINEAGE),
+   createDataWithPangoLineageValue("id_1", SOME_BASE_PANGO_LINEAGE),
+   createDataWithPangoLineageValue("id_2", SOME_SUBLINEAGE),
+   createDataWithPangoLineageNullValue("id_3")
+};
+
+const auto DATABASE_CONFIG = DatabaseConfig{
+   .default_nucleotide_sequence = "segment1",
+   .schema =
+      {.instance_name = "dummy name",
+       .metadata =
+          {{.name = "primaryKey", .type = ValueType::STRING},
+           {.name = "pango_lineage", .type = ValueType::PANGOLINEAGE, .generate_index = true}},
+       .primary_key = "primaryKey"}
+};
+
+const auto REFERENCE_GENOMES = ReferenceGenomes{
+   {{"segment1", "A"}},
+   {{"gene1", "*"}},
+};
+
+const auto ALIAS_LOOKUP = PangoLineageAliasLookup{{{"A", {"P.1"}}}};
```

Comment:
```suggestion
const auto ALIAS_LOOKUP = PangoLineageAliasLookup{{{"A", {"P.1.1.1"}}}};
```
I think there is a certain logic that expects this to have 3 numbers.